{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Lichen Proteome Workbook for python\n",
    "\n",
    "Author: Eric Whisnant\n",
    "\n",
    "Date: 04/02/2025\n",
    "\n",
    "* Other data wrangling was done in the notebook: `lichen-proteome-pilot.qmd`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import os\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(18811, 39)"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Read in the two data frames to work with:\n",
    "\n",
    "combined = pd.read_csv(\"annotated-lichen-proteins.csv\")\n",
    "\n",
    "peltigera = pd.read_csv(\"annotated-peltigera-proteins.csv\", low_memory=False)\n",
    "\n",
    "peltigera.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Here is what needs to happen:\n",
    "\n",
    "1. Group the rows by metataxa to create seperate dataframes for each metataxa.\n",
    "\n",
    "* Do this for both dataframes.\n",
    "\n",
    "DONE\n",
    "\n",
    "2. Remove columns that are unneccessary\n",
    "\n",
    "DONE\n",
    "\n",
    "3. For each `signature_number` extract the sequence and create a fasta file that makes:\n",
    "\n",
    "\"> <match name> | <gene name> | <protein_name>\" as the start of the header, followed by the sequence:\n",
    "\n",
    "DONE\n",
    "\n",
    "3. Create a consensus profile for each `signature_number` for each of the descriptor columns.\n",
    "\n",
    "4. Then I need to\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Combined dataframe (skip ahead for Peligera)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\ericd\\AppData\\Local\\Temp\\ipykernel_35144\\2954770341.py:13: FutureWarning: When grouping with a length-1 list-like, you will need to pass a length-1 tuple to get_group in a future version of pandas. Pass `(name,)` instead of `name` to silence this warning.\n",
      "  [gb_metataxa_comb.get_group(x) for x in gb_metataxa_comb.groups]\n",
      "C:\\Users\\ericd\\AppData\\Local\\Temp\\ipykernel_35144\\2954770341.py:16: FutureWarning: When grouping with a length-1 list-like, you will need to pass a length-1 tuple to get_group in a future version of pandas. Pass `(name,)` instead of `name` to silence this warning.\n",
      "  phycobiont_comb = gb_metataxa_comb.get_group(\"Phycobiont\")\n",
      "C:\\Users\\ericd\\AppData\\Local\\Temp\\ipykernel_35144\\2954770341.py:17: FutureWarning: When grouping with a length-1 list-like, you will need to pass a length-1 tuple to get_group in a future version of pandas. Pass `(name,)` instead of `name` to silence this warning.\n",
      "  mycobiont_comb = gb_metataxa_comb.get_group(\"Mycobiont\")\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>signature_number</th>\n",
       "      <th>metataxa</th>\n",
       "      <th>database</th>\n",
       "      <th>matched_organisms</th>\n",
       "      <th>protein_name</th>\n",
       "      <th>protein_description.x</th>\n",
       "      <th>nmbr_precursors_experiment_wide</th>\n",
       "      <th>nmbr_precursors_controls</th>\n",
       "      <th>nmbr_precursors_cladonia_sample</th>\n",
       "      <th>nmbr_precursors_peltigera_sample</th>\n",
       "      <th>...</th>\n",
       "      <th>GO_ID</th>\n",
       "      <th>sequence</th>\n",
       "      <th>gene_encoded_by</th>\n",
       "      <th>Ensembl</th>\n",
       "      <th>EnsemblFungi</th>\n",
       "      <th>EnsemblPlants</th>\n",
       "      <th>KEGG</th>\n",
       "      <th>eggNOG</th>\n",
       "      <th>EMBL</th>\n",
       "      <th>RefSeq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>8</td>\n",
       "      <td>Mycobiont</td>\n",
       "      <td>Lecanaromycetes</td>\n",
       "      <td>Graphis pulverulenta</td>\n",
       "      <td>A0A023W1D8_9LECA</td>\n",
       "      <td>Elongation factor 1-alpha (Fragment)</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>GO:0003746; GO:0003924; GO:0005525</td>\n",
       "      <td>DGQTREHALLAYTLGVKQLIVAVNKMDTTKWSEDRFNEIVKEVSNF...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>KJ441074;</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>308</th>\n",
       "      <td>42</td>\n",
       "      <td>Mycobiont</td>\n",
       "      <td>Lecanaromycetes</td>\n",
       "      <td>Alectoria fallacina</td>\n",
       "      <td>A0A8H3FNR3_9LECA</td>\n",
       "      <td>GST C-terminal domain-containing protein</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>GO:0004364; GO:0005737</td>\n",
       "      <td>MPNHPPTDGSWRHQLGTPSFPLEPDRYHLYVGLFCPFAHRVIIARE...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CAJPDR010000224;</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>309</th>\n",
       "      <td>42</td>\n",
       "      <td>Mycobiont</td>\n",
       "      <td>Lecanaromycetes</td>\n",
       "      <td>Letharia lupina</td>\n",
       "      <td>A0A8H6C7B5_9LECA</td>\n",
       "      <td>GST C-terminal domain-containing protein</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>GO:0004364; GO:0005737</td>\n",
       "      <td>MPNHPPTDGSWRHQLGTPSFPVEPDRYHLYVGLFCPFAQRAIIARE...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>JACCJB010000024;</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>310</th>\n",
       "      <td>42</td>\n",
       "      <td>Mycobiont</td>\n",
       "      <td>Lecanaromycetes</td>\n",
       "      <td>Letharia columbiana</td>\n",
       "      <td>A0A8H6L172_9LECA</td>\n",
       "      <td>GST C-terminal domain-containing protein</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>GO:0004364; GO:0005737</td>\n",
       "      <td>MPNHPPTDGSWRHQLGTPSFPVEPDRYYLYVGLFCPFAQRAIIARE...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>JACCJC010000054;</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>312</th>\n",
       "      <td>42</td>\n",
       "      <td>Mycobiont</td>\n",
       "      <td>Lecanaromycetes</td>\n",
       "      <td>Lepraria neglecta</td>\n",
       "      <td>A0AAD9ZFV9_9LECA</td>\n",
       "      <td>GST N-terminal domain-containing protein</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>GO:0004364; GO:0005737</td>\n",
       "      <td>MPNYPPSDGSWRHTLGSPSFPVEPDRYHLYVGLFCPFAHRVIVTRE...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>JASNWA010000003;</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     signature_number   metataxa         database     matched_organisms  \\\n",
       "78                  8  Mycobiont  Lecanaromycetes  Graphis pulverulenta   \n",
       "308                42  Mycobiont  Lecanaromycetes   Alectoria fallacina   \n",
       "309                42  Mycobiont  Lecanaromycetes       Letharia lupina   \n",
       "310                42  Mycobiont  Lecanaromycetes   Letharia columbiana   \n",
       "312                42  Mycobiont  Lecanaromycetes     Lepraria neglecta   \n",
       "\n",
       "         protein_name                     protein_description.x  \\\n",
       "78   A0A023W1D8_9LECA      Elongation factor 1-alpha (Fragment)   \n",
       "308  A0A8H3FNR3_9LECA  GST C-terminal domain-containing protein   \n",
       "309  A0A8H6C7B5_9LECA  GST C-terminal domain-containing protein   \n",
       "310  A0A8H6L172_9LECA  GST C-terminal domain-containing protein   \n",
       "312  A0AAD9ZFV9_9LECA  GST N-terminal domain-containing protein   \n",
       "\n",
       "     nmbr_precursors_experiment_wide  nmbr_precursors_controls  \\\n",
       "78                                 1                         0   \n",
       "308                                1                         0   \n",
       "309                                1                         0   \n",
       "310                                1                         0   \n",
       "312                                1                         0   \n",
       "\n",
       "     nmbr_precursors_cladonia_sample  nmbr_precursors_peltigera_sample  ...  \\\n",
       "78                                 1                                 1  ...   \n",
       "308                                1                                 0  ...   \n",
       "309                                1                                 0  ...   \n",
       "310                                1                                 0  ...   \n",
       "312                                1                                 0  ...   \n",
       "\n",
       "                                  GO_ID  \\\n",
       "78   GO:0003746; GO:0003924; GO:0005525   \n",
       "308              GO:0004364; GO:0005737   \n",
       "309              GO:0004364; GO:0005737   \n",
       "310              GO:0004364; GO:0005737   \n",
       "312              GO:0004364; GO:0005737   \n",
       "\n",
       "                                              sequence  gene_encoded_by  \\\n",
       "78   DGQTREHALLAYTLGVKQLIVAVNKMDTTKWSEDRFNEIVKEVSNF...              NaN   \n",
       "308  MPNHPPTDGSWRHQLGTPSFPLEPDRYHLYVGLFCPFAHRVIIARE...              NaN   \n",
       "309  MPNHPPTDGSWRHQLGTPSFPVEPDRYHLYVGLFCPFAQRAIIARE...              NaN   \n",
       "310  MPNHPPTDGSWRHQLGTPSFPVEPDRYYLYVGLFCPFAQRAIIARE...              NaN   \n",
       "312  MPNYPPSDGSWRHTLGSPSFPVEPDRYHLYVGLFCPFAHRVIVTRE...              NaN   \n",
       "\n",
       "    Ensembl EnsemblFungi EnsemblPlants  KEGG eggNOG              EMBL RefSeq  \n",
       "78      NaN          NaN           NaN   NaN    NaN         KJ441074;    NaN  \n",
       "308     NaN          NaN           NaN   NaN    NaN  CAJPDR010000224;    NaN  \n",
       "309     NaN          NaN           NaN   NaN    NaN  JACCJB010000024;    NaN  \n",
       "310     NaN          NaN           NaN   NaN    NaN  JACCJC010000054;    NaN  \n",
       "312     NaN          NaN           NaN   NaN    NaN  JASNWA010000003;    NaN  \n",
       "\n",
       "[5 rows x 31 columns]"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined.columns\n",
    "\n",
    "## Drop columns (this need only be done once):\n",
    "combined_clean = combined.drop(columns=[\"genera\", \"gene_name.x\", \"entry\", \"sequence_uncertainty\", \"protein_description.y\", \"Organism\"])\n",
    "\n",
    "# Combined now has these columns dropped\n",
    "combined_clean.head()\n",
    "\n",
    "# Group by metataxa:\n",
    "gb_metataxa_comb = combined_clean.groupby([\"metataxa\"])\n",
    "\n",
    "# The way to grab the groups is to use the get_group method:\n",
    "[gb_metataxa_comb.get_group(x) for x in gb_metataxa_comb.groups]\n",
    "\n",
    "# The groups are:\n",
    "phycobiont_comb = gb_metataxa_comb.get_group(\"Phycobiont\")\n",
    "mycobiont_comb = gb_metataxa_comb.get_group(\"Mycobiont\")\n",
    "\n",
    "phycobiont_comb.head()\n",
    "mycobiont_comb.head()\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now that they are split apart, lets try and create unique fasta files for each signature_num"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\ericd\\AppData\\Local\\Temp\\ipykernel_35144\\1553858018.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mycobiont_comb[\"basename\"] = mycobiont_comb[[\"signature_number\",\"metataxa\", \"matched_organisms\", \"protein_name\" , \"protein_description.x\" ]].astype(str).agg(\"|\".join, axis=1)\n",
      "C:\\Users\\ericd\\AppData\\Local\\Temp\\ipykernel_35144\\1553858018.py:5: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mycobiont_comb[\"basename\"] = mycobiont_comb[\"basename\"].str.replace(\" \", \"_\")\n",
      "C:\\Users\\ericd\\AppData\\Local\\Temp\\ipykernel_35144\\1553858018.py:7: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  phycobiont_comb[\"basename\"] = phycobiont_comb[[\"signature_number\",\"metataxa\", \"matched_organisms\", \"protein_name\" , \"protein_description.x\" ]].astype(str).agg(\"|\".join, axis=1)\n",
      "C:\\Users\\ericd\\AppData\\Local\\Temp\\ipykernel_35144\\1553858018.py:8: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  phycobiont_comb[\"basename\"] = phycobiont_comb[\"basename\"].str.replace(\" \", \"_\")\n"
     ]
    }
   ],
   "source": [
    "# Create fasta files for the signature numbers\n",
    "\n",
    "# Create a \"basename\" for the fasta entry\n",
    "mycobiont_comb[\"basename\"] = mycobiont_comb[[\"signature_number\",\"metataxa\", \"matched_organisms\", \"protein_name\" , \"protein_description.x\" ]].astype(str).agg(\"|\".join, axis=1)\n",
    "mycobiont_comb[\"basename\"] = mycobiont_comb[\"basename\"].str.replace(\" \", \"_\")\n",
    "\n",
    "phycobiont_comb[\"basename\"] = phycobiont_comb[[\"signature_number\",\"metataxa\", \"matched_organisms\", \"protein_name\" , \"protein_description.x\" ]].astype(str).agg(\"|\".join, axis=1)\n",
    "phycobiont_comb[\"basename\"] = phycobiont_comb[\"basename\"].str.replace(\" \", \"_\")\n",
    "\n",
    "mycobiont_comb\n",
    "\n",
    "# Group by signature number\n",
    "gb_sig_num_phycobiont_comb = phycobiont_comb.groupby([\"signature_number\"])\n",
    "gb_sig_num_mycobiont_comb = mycobiont_comb.groupby([\"signature_number\"])\n",
    "\n",
    "mycobiont_seq = mycobiont_comb[[\"signature_number\", \"basename\", \"sequence\"]].copy()\n",
    "gb_sig_num_mycobiont_seq = mycobiont_seq.groupby([\"signature_number\"])\n",
    "\n",
    "phycobiont_seq = phycobiont_comb[[\"signature_number\", \"basename\", \"sequence\"]].copy()\n",
    "gb_sig_num_phycobiont_seq = phycobiont_seq.groupby([\"signature_number\"])\n",
    "\n",
    "\n",
    "# Create a directory for the mycobiont FASTA files if it doesn't exist\n",
    "output_dir_mycobiont = \"combined/mycobiont_fasta_files\"\n",
    "os.makedirs(output_dir_mycobiont, exist_ok=True)\n",
    "\n",
    "# Create a directory for the phycobiont FASTA files if it doesn't exist\n",
    "output_dir_phycobiont = \"combined/phycobiont_fasta_files\"\n",
    "os.makedirs(output_dir_phycobiont, exist_ok=True)\n",
    "\n",
    "# Create a FASTA file for each signature number\n",
    "for sig_num, group in gb_sig_num_mycobiont_seq:\n",
    "    with open(os.path.join(output_dir, f\"signature_{sig_num}.faa\"), \"w\") as f:\n",
    "        for index, row in group.iterrows():\n",
    "            f.write(f\">{row['basename']}\\n{row['sequence']}\\n\")\n",
    "\n",
    "# Create a FASTA file for each signature number\n",
    "for sig_num, group in gb_sig_num_phycobiont_seq:\n",
    "    with open(os.path.join(output_dir_phycobiont, f\"signature_{sig_num}.faa\"), \"w\") as f:\n",
    "        for index, row in group.iterrows():\n",
    "            f.write(f\">{row['basename']}\\n{row['sequence']}\\n\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>signature_number</th>\n",
       "      <th>basename</th>\n",
       "      <th>sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1|Phycobiont|Auxenochlorella_protothecoides|A0...</td>\n",
       "      <td>MVKKKINGIKKVKIGLTSPREIQEWGERSLLDGTIVGEVSSWETVN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1|Phycobiont|Chlorella_variabilis|A0A075WQ93_C...</td>\n",
       "      <td>MNIQKARVFEKLEIGVASPKQIRHWAERFLPNGDTVGEVTSWETVN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1|Phycobiont|Auxenochlorella_protothecoides|A0...</td>\n",
       "      <td>VKIGLTSPREIQEWGERSLLDGTIVGEVSSWETVNYKTLKPEMGGL...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1|Phycobiont|Prasiolopsis_sp._SAG_84.81|A0A097...</td>\n",
       "      <td>MSKSLIGHTSTQKFSNNRHLFKPLKKSSEAQIDSINIGLASPIRIL...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>1|Phycobiont|Edaphochlorella_mirabilis|A0A097K...</td>\n",
       "      <td>MTKQNTFKSIQVESIRIGLASPDCIRKWAERTLPNGKIIGKVTSRE...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15997</th>\n",
       "      <td>9737</td>\n",
       "      <td>9737|Phycobiont|Prototheca_wickerhamii|Q5IWX0_...</td>\n",
       "      <td>CHMTGVAAQVVASPSRCLFASSQRTSKTASTSRCPSSRRWASLTTP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15998</th>\n",
       "      <td>9738</td>\n",
       "      <td>9738|Phycobiont|Chlorella_ellipsoidea|Q6SKR1_C...</td>\n",
       "      <td>ELNLDDVGYDDVGGVRKQMAQIRELVELPLRHPQLFKTIGVKPPKG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16001</th>\n",
       "      <td>9741</td>\n",
       "      <td>9741|Phycobiont|Trebouxia_jamesii|Q9LL42_9CHLO...</td>\n",
       "      <td>PEANREKMTQIMFETFNTPAMYVAIQAVLSLYASGRTTGIVLDSGD...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16003</th>\n",
       "      <td>9743</td>\n",
       "      <td>9743|Phycobiont|Stichococcus_sp._BCP-ZNP2-VF4|...</td>\n",
       "      <td>TYYTPDYQVKETDVLAAFRMTPQSGVPAEECGAAVAAESSTGTWTT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16004</th>\n",
       "      <td>9744</td>\n",
       "      <td>9744|Phycobiont|Prasiolales_sp._Tisnes_1|W5S92...</td>\n",
       "      <td>RCYDIESVPGEENQYIAYVAYPLDLFEEGSVTNVLTSIVGNVFGFK...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5397 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       signature_number                                           basename  \\\n",
       "0                     1  1|Phycobiont|Auxenochlorella_protothecoides|A0...   \n",
       "1                     1  1|Phycobiont|Chlorella_variabilis|A0A075WQ93_C...   \n",
       "2                     1  1|Phycobiont|Auxenochlorella_protothecoides|A0...   \n",
       "3                     1  1|Phycobiont|Prasiolopsis_sp._SAG_84.81|A0A097...   \n",
       "4                     1  1|Phycobiont|Edaphochlorella_mirabilis|A0A097K...   \n",
       "...                 ...                                                ...   \n",
       "15997              9737  9737|Phycobiont|Prototheca_wickerhamii|Q5IWX0_...   \n",
       "15998              9738  9738|Phycobiont|Chlorella_ellipsoidea|Q6SKR1_C...   \n",
       "16001              9741  9741|Phycobiont|Trebouxia_jamesii|Q9LL42_9CHLO...   \n",
       "16003              9743  9743|Phycobiont|Stichococcus_sp._BCP-ZNP2-VF4|...   \n",
       "16004              9744  9744|Phycobiont|Prasiolales_sp._Tisnes_1|W5S92...   \n",
       "\n",
       "                                                sequence  \n",
       "0      MVKKKINGIKKVKIGLTSPREIQEWGERSLLDGTIVGEVSSWETVN...  \n",
       "1      MNIQKARVFEKLEIGVASPKQIRHWAERFLPNGDTVGEVTSWETVN...  \n",
       "2      VKIGLTSPREIQEWGERSLLDGTIVGEVSSWETVNYKTLKPEMGGL...  \n",
       "3      MSKSLIGHTSTQKFSNNRHLFKPLKKSSEAQIDSINIGLASPIRIL...  \n",
       "4      MTKQNTFKSIQVESIRIGLASPDCIRKWAERTLPNGKIIGKVTSRE...  \n",
       "...                                                  ...  \n",
       "15997  CHMTGVAAQVVASPSRCLFASSQRTSKTASTSRCPSSRRWASLTTP...  \n",
       "15998  ELNLDDVGYDDVGGVRKQMAQIRELVELPLRHPQLFKTIGVKPPKG...  \n",
       "16001  PEANREKMTQIMFETFNTPAMYVAIQAVLSLYASGRTTGIVLDSGD...  \n",
       "16003  TYYTPDYQVKETDVLAAFRMTPQSGVPAEECGAAVAAESSTGTWTT...  \n",
       "16004  RCYDIESVPGEENQYIAYVAYPLDLFEEGSVTNVLTSIVGNVFGFK...  \n",
       "\n",
       "[5397 rows x 3 columns]"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phycobiont_seq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Create fasta files for the peltigera "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\ericd\\AppData\\Local\\Temp\\ipykernel_35144\\3246787746.py:7: FutureWarning: When grouping with a length-1 list-like, you will need to pass a length-1 tuple to get_group in a future version of pandas. Pass `(name,)` instead of `name` to silence this warning.\n",
      "  [gb_metataxa_pelt.get_group(x) for x in gb_metataxa_pelt.groups]\n",
      "C:\\Users\\ericd\\AppData\\Local\\Temp\\ipykernel_35144\\3246787746.py:10: FutureWarning: When grouping with a length-1 list-like, you will need to pass a length-1 tuple to get_group in a future version of pandas. Pass `(name,)` instead of `name` to silence this warning.\n",
      "  cyanobiont_pelt = gb_metataxa_pelt.get_group(\"Cyanobiont\")\n",
      "C:\\Users\\ericd\\AppData\\Local\\Temp\\ipykernel_35144\\3246787746.py:11: FutureWarning: When grouping with a length-1 list-like, you will need to pass a length-1 tuple to get_group in a future version of pandas. Pass `(name,)` instead of `name` to silence this warning.\n",
      "  mycobiont_pelt = gb_metataxa_pelt.get_group(\"Mycobiont\")\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>signature_number</th>\n",
       "      <th>metataxa</th>\n",
       "      <th>database</th>\n",
       "      <th>matched_organisms</th>\n",
       "      <th>protein_name</th>\n",
       "      <th>protein_description.x</th>\n",
       "      <th>nmbr_precursors_experiment_wide</th>\n",
       "      <th>nmbr_precursors_controls</th>\n",
       "      <th>nmbr_precursors_peltigera</th>\n",
       "      <th>relative_abundance_controls</th>\n",
       "      <th>...</th>\n",
       "      <th>RefSeq</th>\n",
       "      <th>CAZy</th>\n",
       "      <th>MEROPS</th>\n",
       "      <th>UniLectin</th>\n",
       "      <th>Ensembl</th>\n",
       "      <th>EnsemblFungi</th>\n",
       "      <th>EnsemblPlants</th>\n",
       "      <th>EnsemblBacteria</th>\n",
       "      <th>KEGG</th>\n",
       "      <th>eggNOG</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>238</th>\n",
       "      <td>2</td>\n",
       "      <td>Mycobiont</td>\n",
       "      <td>Lecanaromycetes</td>\n",
       "      <td>Graphis pulverulenta</td>\n",
       "      <td>A0A023W1D8_9LECA</td>\n",
       "      <td>Elongation factor 1-alpha (Fragment)</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>239</th>\n",
       "      <td>3</td>\n",
       "      <td>Mycobiont</td>\n",
       "      <td>Lecanaromycetes</td>\n",
       "      <td>Peltigera dolichorrhiza</td>\n",
       "      <td>A0A075QJ18_9LECA</td>\n",
       "      <td>Elongation factor 2 (Fragment)</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>240</th>\n",
       "      <td>3</td>\n",
       "      <td>Mycobiont</td>\n",
       "      <td>Lecanaromycetes</td>\n",
       "      <td>Peltigera dolichorrhiza</td>\n",
       "      <td>A0A1D9IVW2_9LECA</td>\n",
       "      <td>Elongation factor 2 (Fragment)</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241</th>\n",
       "      <td>3</td>\n",
       "      <td>Mycobiont</td>\n",
       "      <td>Lecanaromycetes</td>\n",
       "      <td>Peltigera hymenina</td>\n",
       "      <td>A0A1D9IVW7_9LECA</td>\n",
       "      <td>Elongation factor 2 (Fragment)</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>242</th>\n",
       "      <td>3</td>\n",
       "      <td>Mycobiont</td>\n",
       "      <td>Lecanaromycetes</td>\n",
       "      <td>Peltigera hymenina</td>\n",
       "      <td>A0A1D9IVW9_9LECA</td>\n",
       "      <td>Elongation factor 2 (Fragment)</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 33 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     signature_number   metataxa         database        matched_organisms  \\\n",
       "238                 2  Mycobiont  Lecanaromycetes     Graphis pulverulenta   \n",
       "239                 3  Mycobiont  Lecanaromycetes  Peltigera dolichorrhiza   \n",
       "240                 3  Mycobiont  Lecanaromycetes  Peltigera dolichorrhiza   \n",
       "241                 3  Mycobiont  Lecanaromycetes       Peltigera hymenina   \n",
       "242                 3  Mycobiont  Lecanaromycetes       Peltigera hymenina   \n",
       "\n",
       "         protein_name                 protein_description.x  \\\n",
       "238  A0A023W1D8_9LECA  Elongation factor 1-alpha (Fragment)   \n",
       "239  A0A075QJ18_9LECA        Elongation factor 2 (Fragment)   \n",
       "240  A0A1D9IVW2_9LECA        Elongation factor 2 (Fragment)   \n",
       "241  A0A1D9IVW7_9LECA        Elongation factor 2 (Fragment)   \n",
       "242  A0A1D9IVW9_9LECA        Elongation factor 2 (Fragment)   \n",
       "\n",
       "     nmbr_precursors_experiment_wide  nmbr_precursors_controls  \\\n",
       "238                                1                         0   \n",
       "239                                1                         0   \n",
       "240                                1                         0   \n",
       "241                                1                         0   \n",
       "242                                1                         0   \n",
       "\n",
       "     nmbr_precursors_peltigera  relative_abundance_controls  ...  RefSeq CAZy  \\\n",
       "238                          1                          NaN  ...     NaN  NaN   \n",
       "239                          1                          NaN  ...     NaN  NaN   \n",
       "240                          1                          NaN  ...     NaN  NaN   \n",
       "241                          1                          NaN  ...     NaN  NaN   \n",
       "242                          1                          NaN  ...     NaN  NaN   \n",
       "\n",
       "    MEROPS UniLectin  Ensembl EnsemblFungi EnsemblPlants EnsemblBacteria KEGG  \\\n",
       "238    NaN       NaN      NaN          NaN           NaN             NaN  NaN   \n",
       "239    NaN       NaN      NaN          NaN           NaN             NaN  NaN   \n",
       "240    NaN       NaN      NaN          NaN           NaN             NaN  NaN   \n",
       "241    NaN       NaN      NaN          NaN           NaN             NaN  NaN   \n",
       "242    NaN       NaN      NaN          NaN           NaN             NaN  NaN   \n",
       "\n",
       "    eggNOG  \n",
       "238    NaN  \n",
       "239    NaN  \n",
       "240    NaN  \n",
       "241    NaN  \n",
       "242    NaN  \n",
       "\n",
       "[5 rows x 33 columns]"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "peltigera.head()\n",
    "\n",
    "peltigera_cleaned = peltigera.drop(columns=[\"genera\", \"gene_name.x\", \"entry\", \"sequence_uncertainty\", \"protein_description.y\", \"Organism\"])\n",
    "\n",
    "# Group by metataxa:\n",
    "gb_metataxa_pelt = peltigera.groupby([\"metataxa\"])\n",
    "\n",
    "# The way to grab the groups is to use the get_group method:\n",
    "[gb_metataxa_pelt.get_group(x) for x in gb_metataxa_pelt.groups]\n",
    "\n",
    "# The groups are:\n",
    "cyanobiont_pelt = gb_metataxa_pelt.get_group(\"Cyanobiont\")\n",
    "mycobiont_pelt = gb_metataxa_pelt.get_group(\"Mycobiont\")\n",
    "\n",
    "cyanobiont_pelt.head()\n",
    "mycobiont_pelt.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\ericd\\AppData\\Local\\Temp\\ipykernel_35144\\1525911803.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mycobiont_pelt[\"basename\"] = mycobiont_pelt[[\"signature_number\", \"metataxa\", \"matched_organisms\", \"protein_name\", \"protein_description.x\"]].astype(str).agg(\"|\".join, axis=1)\n",
      "C:\\Users\\ericd\\AppData\\Local\\Temp\\ipykernel_35144\\1525911803.py:5: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mycobiont_pelt[\"basename\"] = mycobiont_pelt[\"basename\"].str.replace(\" \", \"_\")\n",
      "C:\\Users\\ericd\\AppData\\Local\\Temp\\ipykernel_35144\\1525911803.py:7: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  cyanobiont_pelt[\"basename\"] = cyanobiont_pelt[[\"signature_number\", \"metataxa\", \"matched_organisms\", \"protein_name\", \"protein_description.x\"]].astype(str).agg(\"|\".join, axis=1)\n",
      "C:\\Users\\ericd\\AppData\\Local\\Temp\\ipykernel_35144\\1525911803.py:8: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  cyanobiont_pelt[\"basename\"] = cyanobiont_pelt[\"basename\"].str.replace(\" \", \"_\")\n"
     ]
    }
   ],
   "source": [
    "# Create fasta files for the signature numbers\n",
    "\n",
    "# Create a \"basename\" for the fasta entry\n",
    "mycobiont_pelt[\"basename\"] = mycobiont_pelt[[\"signature_number\", \"metataxa\", \"matched_organisms\", \"protein_name\", \"protein_description.x\"]].astype(str).agg(\"|\".join, axis=1)\n",
    "mycobiont_pelt[\"basename\"] = mycobiont_pelt[\"basename\"].str.replace(\" \", \"_\")\n",
    "\n",
    "cyanobiont_pelt[\"basename\"] = cyanobiont_pelt[[\"signature_number\", \"metataxa\", \"matched_organisms\", \"protein_name\", \"protein_description.x\"]].astype(str).agg(\"|\".join, axis=1)\n",
    "cyanobiont_pelt[\"basename\"] = cyanobiont_pelt[\"basename\"].str.replace(\" \", \"_\")\n",
    "\n",
    "mycobiont_pelt\n",
    "\n",
    "# Group by signature number\n",
    "gb_sig_num_cyanobiont_pelt = cyanobiont_pelt.groupby([\"signature_number\"])\n",
    "gb_sig_num_mycobiont_pelt = mycobiont_pelt.groupby([\"signature_number\"])\n",
    "\n",
    "mycobiont_seq_pelt = mycobiont_pelt[[\"signature_number\", \"basename\", \"sequence\"]].copy()\n",
    "gb_sig_num_mycobiont_seq_pelt = mycobiont_seq_pelt.groupby([\"signature_number\"])\n",
    "\n",
    "cyanobiont_seq = cyanobiont_pelt[[\"signature_number\", \"basename\", \"sequence\"]].copy()\n",
    "gb_sig_num_cyanobiont_seq = cyanobiont_seq.groupby([\"signature_number\"])\n",
    "\n",
    "# Create a directory for the mycobiont FASTA files if it doesn't exist\n",
    "output_dir = \"peltigera/mycobiont_fasta_files\"\n",
    "os.makedirs(output_dir, exist_ok=True)\n",
    "\n",
    "# Create a directory for the cyanobiont FASTA files if it doesn't exist\n",
    "output_dir_cyanobiont = \"peltigera/cyanobiont_fasta_files\"\n",
    "os.makedirs(output_dir_cyanobiont, exist_ok=True)\n",
    "\n",
    "# Create a FASTA file for each signature number\n",
    "for sig_num, group in gb_sig_num_mycobiont_seq_pelt:\n",
    "    with open(os.path.join(output_dir, f\"signature_{sig_num}.faa\"), \"w\") as f:\n",
    "        for index, row in group.iterrows():\n",
    "            f.write(f\">{row['basename']}\\n{row['sequence']}\\n\")\n",
    "\n",
    "# Create a FASTA file for each signature number\n",
    "for sig_num, group in gb_sig_num_cyanobiont_seq:\n",
    "    with open(os.path.join(output_dir_cyanobiont, f\"signature_{sig_num}.faa\"), \"w\") as f:\n",
    "        for index, row in group.iterrows():\n",
    "            f.write(f\">{row['basename']}\\n{row['sequence']}\\n\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Let us make a consensus file with all of the signatures"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Now do the peltigera specific \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(18811, 33)\n",
      "(12074, 33)\n",
      "(6631, 33)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\ericd\\AppData\\Local\\Temp\\ipykernel_35144\\4273829176.py:6: FutureWarning: When grouping with a length-1 list-like, you will need to pass a length-1 tuple to get_group in a future version of pandas. Pass `(name,)` instead of `name` to silence this warning.\n",
      "  [gb_metataxa_pelt.get_group(x) for x in gb_metataxa_pelt.groups]\n",
      "C:\\Users\\ericd\\AppData\\Local\\Temp\\ipykernel_35144\\4273829176.py:8: FutureWarning: When grouping with a length-1 list-like, you will need to pass a length-1 tuple to get_group in a future version of pandas. Pass `(name,)` instead of `name` to silence this warning.\n",
      "  cyanobiont_pelt = gb_metataxa_pelt.get_group(\"Cyanobiont\")\n",
      "C:\\Users\\ericd\\AppData\\Local\\Temp\\ipykernel_35144\\4273829176.py:10: FutureWarning: When grouping with a length-1 list-like, you will need to pass a length-1 tuple to get_group in a future version of pandas. Pass `(name,)` instead of `name` to silence this warning.\n",
      "  mycobiont_pelt = gb_metataxa_pelt.get_group(\"Mycobiont\")\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "18705"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# peltigera = peltigera.drop(columns=[\"genera\", \"gene_name.x\", \"entry\", \"sequence_uncertainty\", \"protein_description.y\", \"Organism\"])\n",
    "\n",
    "\n",
    "gb_metataxa_pelt = peltigera.groupby([\"metataxa\"])\n",
    "\n",
    "[gb_metataxa_pelt.get_group(x) for x in gb_metataxa_pelt.groups]\n",
    "\n",
    "cyanobiont_pelt = gb_metataxa_pelt.get_group(\"Cyanobiont\")\n",
    "\n",
    "mycobiont_pelt = gb_metataxa_pelt.get_group(\"Mycobiont\")\n",
    "\n",
    "# nan_pelt = gb_metataxa_pelt.get_group(\"NaN\")\n",
    "\n",
    "print(peltigera.shape)\n",
    "\n",
    "cyanobiont_pelt.head()\n",
    "\n",
    "print(cyanobiont_pelt.shape)\n",
    "\n",
    "mycobiont_pelt.head()\n",
    "\n",
    "print(mycobiont_pelt.shape)\n",
    "\n",
    "# Note: the missing rows after splitting the data frames are bc the proteins are coming from cow, human, or non lichen source\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
